### PR TITLE
Optimize the release process instructions slightly

### PR DESCRIPTION
**Pull request by benjaoming**
### Summary

Noticed that we haven't been following a release process for the latest releases. This can hopefully help to shed light on remaining slip-ups.

### Reviewer guidance

Please add in your own optimizations :)

### References

#2762

----

### Contributor Checklist

- [x] PR has the correct target milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, it has been assigned or requests review from someone and labeled as 'needs review'
- [x] PR has been fully tested manually
- [x] Documentation is updated
- [x] External dependencies files were updated (`yarn` and `pip`)
- [x] Link to diff of internal dependency change is included
- [x] Screenshots of any front-end changes are in the PR description
- [x] PR does not introduce [accessibility regressions](http://kolibri.readthedocs.io/en/develop/dev/manual_testing.html#accessibility-a11y-testing)
- [x] CHANGELOG.rst is updated for high-level changes
- [x] You've added yourself to AUTHORS.rst if you're not there

### Reviewer Checklist

- [ ] Automated test coverage is satisfactory
- [ ] PR has been fully tested manually

### CODE REVIEW

File: docs-developer/release_process.rst [hunk 1]
```diff
@@ -85,25 +84,20 @@ Ensure bugfixes from internal depencies have propagated
 Some issues in Kolibri arise due to our integration of internally produced, but external to Kolibri, packages, such as kolibri-exercise-perseus-renderer, iceqube, and morango. If any of these kinds of dependencies have been updated to fix issues for this milestone, then the dependency version should have been updated.
 
 
-Edit the VERSION file
-~~~~~~~~~~~~~~~~~~~~~
```
Comment (benjaoming) [inline docs-developer/release_process.rst:89]: That heading was a bit of a left-over :)

File: docs-developer/release_process.rst [hunk 2]
```diff
@@ -85,36 +84,33 @@ Ensure bugfixes from internal depencies have propagated
...
+Bump the version
+~~~~~~~~~~~~~~~~
 
-Current practice is to bump ``kolibri.VERSION`` before tagging a release. You are allowed to have a newer version in ``kolibri.VERSION``, but you are not allowed to add the tag before actually bumping ``kolibri.VERSION``.
+.. note:: This applies to all releases, both stable and pre-releases. Anything that's released should be tracked in a release branch.
```
Comment (rtibbles) [inline docs-developer/release_process.rst:90]: Note: this was not what either @indirectlylit or myself had understood about the design of the versioning system. The PR I opened yesterday was to try to bring the versioning system into line with what we had expected.

**Comment by benjaoming** [comment]
No you're right, we were supposed to be able to version pre-releases straight from tags.

Of course, this time, we tripped over the good old `git describe` issue :tada: :smile:

**Comment by benjaoming** [comment]
Am not interested in opening up any discussion on versioning schemes, so closing this for now.

File: docs-developer/release_process.rst (hunk 2 above)
Comment (benjaoming) [reply docs-developer/release_process.rst:90]: I keep re-learning this for some reason :smile: 

But it's fine, let's leave it.. I can't see making non-tracked pre-releases going all that wrong, after all the tag is there.

**Pull request closed without merging**